app/controllers/chargeB/CheckYourAnswersController.scala

```diff
@@ -47,24 +46,29 @@ class CheckYourAnswersController @Inject()(config: FrontendAppConfig,
                                            renderer: Renderer
                                           )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport with NunjucksSupport {
 
-  def onPageLoad(srn: String): Action[AnyContent] = (identify andThen getData andThen allowAccess(srn) andThen requireData).async {
+  def onPageLoad(srn: String): Action[AnyContent] = (identify andThen getData(srn) andThen allowAccess(srn) andThen requireData).async {
     implicit request =>
       DataRetrievals.cyaChargeGeneric(ChargeBDetailsPage, srn) { (chargeDetails, schemeName) =>
         val helper = new CheckYourAnswersHelper(request.userAnswers, srn)
+        val seqRows = helper.chargeBDetails(chargeDetails)
+
+        val rows = if(request.viewOnly) seqRows.map(_.copy(actions = Nil)) else seqRows
```

Comment:
Might be nicer to pass boolean  (readonly) into helper method and return nil actions if true. This logic could then be in just one file instead of across all the CYA controllers.